src/main/java/com/rabbitmq/client/impl/recovery/RecordedConsumer.java

```diff
@@ -57,7 +57,7 @@ public RecordedConsumer autoAck(boolean value) {
     }
 
     public String recover() throws IOException {
-        this.consumerTag = this.channel.basicConsume(this.queue, this.autoAck, this.consumerTag, false, this.exclusive, this.arguments, this.consumer);
+        this.consumerTag = this.channel.getDelegate().basicConsume(this.queue, this.autoAck, this.consumerTag, false, this.exclusive, this.arguments, this.consumer);
```

Comment:
all the recorded entities besides these 2 use getDelegate(). I think these should 2 as well otherwise it ends up duplicating the RecorcedConsumer/Queue and readding it to the recorded entity collections.